### PR TITLE
PR checklist and Issue template update

### PR DESCRIPTION
**Pull request by leseb**
**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**
- [ ] Documentation has been updated, if necessary.
- [ ] Pending release notes updated with breaking and/or notable changes, if necessary.
- [ ] Upgrade from previous release is tested and upgrade user guide is updated, if necessary.
- [ ] Code generation (`make codegen`) has been run to update object specifications, if necessary.
- [ ] Comments have been added or updated based on the standards set in [CONTRIBUTING.md](../CONTRIBUTING.md#comments)

[skip ci]

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -8,7 +8,10 @@ Thank you for contributing to Rook! -->
 Resolves #
 
 **Checklist:**
+
 - [ ] Documentation has been updated, if necessary.
+- [ ] Unit tests have been added, if necessary.
```
Comment (travisn) [inline .github/PULL_REQUEST_TEMPLATE.md]: How about if we add a link to the dev guide as well and combine the others to a single item?

[ ] Reviewed the developer guide on [Submitting a Pull Request](https://rook.io/docs/rook/v1.0/development-flow.html#submitting-a-pull-request)
[ ] Added unit tests and integration tests, if necessary.

Comment (leseb) [reply .github/PULL_REQUEST_TEMPLATE.md]: The problem with merging the two into one is that often PR are wip so unit tests might be present but integration test coming later.

I'm ok for the dev guide.

**Comment by leseb** [comment]
@travisn updated.

**Comment by galexrt** [comment]
Are you sure we can just remove `.github/ISSUE_TEMPLATE.md`? It is / was the default template used for issues that are created directly without the "issue type selection"  (e.g., https://github.com/rook/rook/issues/new).

**Comment by travisn** [comment]
Is there a way to use that template? I thought github would only use the feature or bug templates now

**Comment by leseb** [comment]
> Is there a way to use that template? I thought github would only use the feature or bug templates now

Me too, also if you click on the file github will tell you that's an old template too.

**Comment by leseb** [comment]
@galexrt final thought?

**Comment by galexrt** [comment]
@leseb Let's go ahead and remove it. If it breaks the new issue form (without going through selection) we simply need to re-add it.

**Comment by galexrt** [comment]
Oh well https://github.com/rook/rook/issues/new

Wanna go ahead and re-add it? :smile:

**Comment by leseb** [comment]
> Oh well https://github.com/rook/rook/issues/new
> 
> Wanna go ahead and re-add it? 😄

Why would people not chose to report a bug or a feature?

**Comment by galexrt** [comment]
> > Oh well https://github.com/rook/rook/issues/new
> > 
> > Wanna go ahead and re-add it? 😄
> 
> Why would people not choose to report a bug or a feature?

Why would people not read the documentation? :wink:


Also people can get to that "state" when clicking the `Don’t see your issue here? Open a regular issue.` in the https://github.com/rook/rook/issues/new/choose form.

**Comment by leseb** [comment]
> > > Oh well https://github.com/rook/rook/issues/new
> > > Wanna go ahead and re-add it? 😄
> > 
> > 
> > Why would people not choose to report a bug or a feature?
> 
> Why would people not read the documentation? 😉
> 
> Also people can get to that "state" when clicking the `Don’t see your issue here? Open a regular issue.` in the https://github.com/rook/rook/issues/new/choose form.

Ok.

**Comment by leseb** [comment]
> > > > Oh well https://github.com/rook/rook/issues/new
> > > > Wanna go ahead and re-add it? 😄
> > > 
> > > 
> > > Why would people not choose to report a bug or a feature?
> > 
> > 
> > Why would people not read the documentation? 😉
> > Also people can get to that "state" when clicking the `Don’t see your issue here? Open a regular issue.` in the https://github.com/rook/rook/issues/new/choose form.
> 
> Ok.

It's just strange to add the same content if people think they are not in the right place.

**Pull request merged**